tests/test_root_versioning.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python
+
+"""
+<Program Name>
+  test_root_versioning.py
```

Comment:
We currently have tests for specific modules (e.g., test_keydb.py) and integration tests / attacks (test_replay_attack.py, i.e., they end in "attack").  Not sure how we should handle the additional tests of this PR.